WordPress/src/main/java/org/wordpress/android/ui/deeplinks/handlers/MediaLinkHandler.kt

```diff
@@ -0,0 +1,57 @@
+package org.wordpress.android.ui.deeplinks.handlers
+
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.ui.deeplinks.DeepLinkNavigator.NavigateAction
+import org.wordpress.android.ui.deeplinks.DeepLinkNavigator.NavigateAction.OpenMedia
+import org.wordpress.android.ui.deeplinks.DeepLinkNavigator.NavigateAction.OpenMediaForSite
+import org.wordpress.android.ui.deeplinks.DeepLinkUriUtils
+import org.wordpress.android.ui.deeplinks.DeepLinkingIntentReceiverViewModel
+import org.wordpress.android.ui.deeplinks.DeepLinkingIntentReceiverViewModel.Companion.HOST_WORDPRESS_COM
+import org.wordpress.android.ui.deeplinks.DeepLinkingIntentReceiverViewModel.Companion.SITE_DOMAIN
+import org.wordpress.android.util.UriWrapper
+import javax.inject.Inject
+
+class MediaLinkHandler
+@Inject constructor(private val deepLinkUriUtils: DeepLinkUriUtils) : DeepLinkHandler {
+    /**
+     * Returns true if the URI looks like `wordpress.com/media`
+     */
+    override fun shouldHandleUrl(uri: UriWrapper): Boolean {
+        return  (uri.host == HOST_WORDPRESS_COM &&
+                uri.pathSegments.firstOrNull() == MEDIA_PATH) || uri.host == MEDIA_PATH
+    }
```

Comment:
There was a typo in the MEDIA_PATH variable, I fixed it with the commit b0b034cf64e4a4cbe3ec94d5b4768b5f29998e22